java/common/src/main/scala/org/typelevel/otel4s/java/context/propagation/convert/AsJavaExtensions.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.java.context
+package propagation
+package convert
+
+import io.opentelemetry.context.propagation.{
+  ContextPropagators => JContextPropagators
+}
+import io.opentelemetry.context.propagation.{TextMapGetter => JTextMapGetter}
+import io.opentelemetry.context.propagation.{
+  TextMapPropagator => JTextMapPropagator
+}
+import org.typelevel.otel4s.context.propagation.ContextPropagators
+import org.typelevel.otel4s.context.propagation.TextMapGetter
+import org.typelevel.otel4s.context.propagation.TextMapPropagator
+
+/** Defines `asJava` extension methods for `TextMapGetter`s and
+  * `TextMapPropagator`s, and `ContextPropagators`, available through
+  * [[org.typelevel.otel4s.java.context.propagation.PropagatorConverters]].
+  */
+trait AsJavaExtensions {
```

Comment:
If we put `implicit class`es right into the `PropagatorConverters` object , we can define them as value classes:

```scala
implicit class TextMapGetterHasAsJava[A](private val getter: TextMapGetter[A]) extends AnyVal
```

In some cases, it could be more optimized in runtime. 

But I don't have a strong opinion on this, to be honest.

Reply:
> But I don't have a strong opinion on this, to be honest.

neither do I. I just structured it the same way `scala.jdk.CollectionConverters` is structured. I have no attachment

Reply:
I say we leave it as is and change it later if needed